.github/workflows/benchmark.yml

```diff
@@ -1,50 +1,260 @@
-name: Benchmark PR vs main
+name: benchmark
 on:
   workflow_dispatch:
   pull_request_review:
     types: [submitted]
   pull_request:
-    branches: [ main ]
+    branches: [main]
     types: [synchronize]
     paths:
-      - 'Sources/*.swift'
-      - .github/workflows/benchmark.yml 
+      - Sources/*.swift
+      - .github/workflows/benchmark.yml
 
 jobs:
-  benchmark-delta-linux:
-    if: github.event.review.state == 'approved'
+  benchmark-vs-thresholds:
+    # Run the job only if it's a manual workflow dispatch, or if this event is a pull-request approval event, or if someone has rerun the job.
+    if: github.event_name == 'workflow_dispatch' || github.event.review.state == 'approved' || github.run_attempt > 1
```

Comment:
updated this condition @gwynne @ptoffy to always run the job if someone has rerun it.